build.gradle

```diff
@@ -13,163 +13,244 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import org.apache.tools.ant.filters.ReplaceTokens
+plugins {
+    id 'com.github.ben-manes.versions' version '0.17.0'
+    id 'com.github.johnrengelman.shadow' version '2.0.2'
+    id 'com.jfrog.bintray' version '1.8.0'
+}
 
-// to build everything:             "gradlew build"
-// to build and upload everything:  "gradlew bintrayUpload"
+tasks.create("build") {
+    group 'build'
+}
 
-plugins {
-    id 'com.jfrog.bintray' version '1.7.3'
-    id 'java'
-    id 'maven-publish'
-    id 'com.github.johnrengelman.shadow' version '2.0.1'
+allprojects {
+    apply plugin: 'maven-publish'
+    apply plugin: 'com.jfrog.bintray'
+    apply plugin: 'com.github.johnrengelman.shadow'
+
+    group 'com.jagrosh'
+    version = '2.0'
+
+    project.ext {
+        jdaVersion = '3.4.0_320'
+        slf4jVersion = '1.7.25'
+        okhttpVersion = '3.9.1'
+        findbugsVersion = '3.0.2'
+
+        jda = { project.dependencies.compileOnly group: 'net.dv8tion', name: 'JDA', version: jdaVersion }
+        slf4j = { project.dependencies.compile group: 'org.slf4j', name: 'slf4j-api', version: slf4jVersion }
+        okhttp = { project.dependencies.compile group: 'com.squareup.okhttp3', name: 'okhttp', version: okhttpVersion }
+        findbugs = { project.dependencies.compile group: 'com.google.code.findbugs', name: 'jsr305', version: findbugsVersion }
```

Comment:
Of the three dependency functions: `slf4j`, `okhttp`, and `findbugs`, JDA compiles at least the first two.
I don't see a need to directly specify them here.

Reply:
It is recommended to declare every dependency explicitly as relying on transitive dependencies can lead to missing dependencies if a user uses another version of a dependency that doesn't have a lib we require anymore. This is especially bad with JD Aas we don't even specify a minimum version and has already happened in the past (keyword Unirest).

Reply:
Yes, every dependency you actually need should be explicitly specified. Don't rely on transitive dependencies.